### ISSUE TITLE
Wrong error location when slicing a slice

### ISSUE THREAD

**Issue by Bunch-of-cells** [rust-lang/rust#101796]
```rust
fn main() {
    let s = &[0, 1];
    let _ = &s[0..4];
}
```

The current output is:

```
thread 'main' panicked at 'range end index 4 out of range for slice of length 2', library/core/src/slice/index.rs:73:5
```

Ideally the output should look like:

```
thread 'main' panicked at 'range end index 4 out of range for slice of length 2', src/main.rs:3:21
```

The error location should be in main.rs, and not library/core/.../index.rs

**Comment by fmease** [comment]
This should already be fixed on nightly via #107059. Can you confirm this? This issue is probably a duplicate of #100696.

**Comment by compiler-errors** [comment]
This is indeed fixed on nightly. Thanks for reporting the issue, closing now!
